server/src/main/java/com/objectcomputing/checkins/services/memberSkill/MemberSkillServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.objectcomputing.checkins.services.memberSkill;
+
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileRepository;
+import com.objectcomputing.checkins.services.skills.SkillRepository;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+public class MemberSkillServiceImpl implements MemberSkillServices {
+
+    @Inject
+    private MemberSkillRepository memberSkillRepository;
+
+    @Inject
+    private MemberProfileRepository memberProfileRepository;
+
+    @Inject
+    private SkillRepository skillRepository;
+
+    public MemberSkill save(MemberSkill memberSkill) {
+        MemberSkill memberSkillRet = null;
+        if (memberSkill != null) {
+            final UUID memberId = memberSkill.getMemberid();
+            final UUID skillId = memberSkill.getSkillid();
+            if (skillId == null || memberId == null) {
```

Comment:
These are annotated as NotNull in the DTO and validated at the controller, so if a request comes in with these values being null a ConstraintViolationException would be thrown before reaching this point. If maintaining this text for the error (as compared to the default text used by ConstraintViolationExceptions) is needed then an error handler will be needed for that as well.

Reply:
I don't think we've decided on that. I like the constraint violation exceptions myself.

Reply:
Replace this with, and make similar changes to update:

```
    public MemberSkill save(@NonNull @NotNull MemberSkill memberSkill) {
        MemberSkill memberSkillRet;
        
        @NotNull
        final UUID memberId = memberSkill.getMemberid();
        @NotNull
        final UUID skillId = memberSkill.getSkillid();
        if (memberSkill.getId() != null) {
            throw new MemberSkillBadArgException(String.format("Found unexpected id %s for member skill", memberSkill.getId()));
        } else if (!memberProfileRepository.findById(memberId).isPresent()) {
            throw new MemberSkillBadArgException(String.format("Member Profile %s doesn't exist", memberId));
        } else if (!skillRepository.findById(skillId).isPresent()) {
            throw new MemberSkillBadArgException(String.format("Skill %s doesn't exist", skillId));
        } else if (memberSkillRepository.findByMemberidAndSkillid(memberSkill.getMemberid(),
                memberSkill.getSkillid()).isPresent()) {
            throw new MemberSkillBadArgException(String.format("Member %s already has this skill %s", memberId, skillId));
        }

        memberSkillRet = memberSkillRepository.save(memberSkill);
        return memberSkillRet;

    }
```

Reply:
I think we should just keep it the way it is, not worth changing it everywhere, especially for these PRs. If you think this should be changed create an issue and open a PR and submit it so we change it in all services in one go, as well as updating the tests.